index.js

```diff
@@ -127,7 +127,7 @@ function generateRewritePrefix (prefix, opts) {
 }
 
 async function httpProxy (fastify, opts) {
-  if (!opts.upstream) {
+  if (!opts.upstream && !(opts.upstream === '' && typeof opts?.replyOptions?.getUpstream === 'function')) {
```

Comment:
Could you avoid the optional chaing? 
We need to target node 10 to ship this feature